doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc

```diff
@@ -0,0 +1,63 @@
+
+[id='business-optimizer-test-constraints-proc_{CONTEXT}']
+= Test the school timetable constraints
+
+To test each constraint of the timetable project in isolation, use a `ConstraintVerifier` in unit tests. Doing this tests each constraint’s corner cases in isolation from the other tests, which lowers maintenance when adding a new constraint with proper test coverage.
+
+This test verifies that the constraint `TimeTableConstraintProvider::roomConflict`, when given three lessons in the same room and two of the lessons have the same timeslot, penalizes with a match weight of 1. So if the constraint weight is `10hard` it reduces the score by `-10hard`.
+
+Procedure
+
+Create the `src/test/java/org/acme/optaplanner/solver/TimeTableConstraintProviderTest.java` class:
++
+[source,java]
+----
+package org.acme.optaplanner.solver;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+import javax.inject.Inject;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.acme.optaplanner.domain.Lesson;
+import org.acme.optaplanner.domain.Room;
+import org.acme.optaplanner.domain.TimeTable;
+import org.acme.optaplanner.domain.Timeslot;
+import org.junit.jupiter.api.Test;
+import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+
+@QuarkusTest
+class TimeTableConstraintProviderTest {
+
+    private static final Room ROOM = new Room("Room1");
+    private static final Timeslot TIMESLOT1 = new Timeslot(DayOfWeek.MONDAY, LocalTime.of(9,0), LocalTime.NOON);
+    private static final Timeslot TIMESLOT2 = new Timeslot(DayOfWeek.TUESDAY, LocalTime.of(9,0), LocalTime.NOON);
+
+    @Inject
+    ConstraintVerifier<TimeTableConstraintProvider, TimeTable> constraintVerifier;
+
+    @Test
+    void roomConflict() {
+        Lesson firstLesson = new Lesson(1, "Subject1", "Teacher1", "Group1");
+        Lesson conflictingLesson = new Lesson(2, "Subject2", "Teacher2", "Group2");
+        Lesson nonConflictingLesson = new Lesson(3, "Subject3", "Teacher3", "Group3");
+
+        firstLesson.setRoom(ROOM);
+        firstLesson.setTimeslot(TIMESLOT1);
+
+        conflictingLesson.setRoom(ROOM);
+        conflictingLesson.setTimeslot(TIMESLOT1);
+
+        nonConflictingLesson.setRoom(ROOM);
+        nonConflictingLesson.setTimeslot(TIMESLOT2);
+
+        constraintVerifier.verifyThat(TimeTableConstraintProvider::roomConflict)
+                .given(firstLesson, conflictingLesson, nonConflictingLesson)
+                .penalizesBy(1);
+    }
+
+}
+----
+
+Notice how `ConstraintVerifier` ignores the constraint weight during testing even if those constraint weights are hard coded in the `ConstraintProvider`. This is because constraint weights change regularly before going into production. This way, constraint weight tweaking does not break the unit tests.
```

Comment:
```suggestion
Notice how `ConstraintVerifier` ignores the constraint weight during testing even if those constraint weights are hardcoded in the `ConstraintProvider`. This is because constraint weights change regularly before going into production. This way, constraint weight tweaking does not break the unit tests.
```

[Style Guide](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#h) says that "hard code" is one word, at least for the verb. Not sure if if it applies in this case.

Reply:
Thanks!